packages/eslint-plugin/docs/rules/no-misused-spread.mdx

```diff
@@ -0,0 +1,141 @@
+---
+description: 'Disallow using the spread operator when it might cause unexpected behavior.'
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-misused-spread** for documentation.
+
+The [spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) (`...`) is a JavaScript
+feature that can be misused in ways not always reported by TypeScript. This rule disallows using the spread operator on types where
+spreading can lead to unexpected behavior.
+
+This rule disallows the spread operator in the following cases:
+
+- Spreading a string into an array
+- Spreading a `Promise` into an object
+- Spreading a function without properties into an object
+- Spreading a class instance or declaration into an object
+- Spreading an iterable (`Map`, `Array`, etc.) into an object
```

Comment:
I think we should have one sentence explanations for each, such as:

```suggestion
- Spreading a `Promise` into an object. You probably meant to `await` it.
- Spreading a function without properties into an object. You probably meant to call it.
- Spreading a class instance into an object. This does not faithfully copy the instance because only its own properties are copied, but the inheritance chain is lost, including all its methods.
- Spreading a class (defined with a `class` declaration or `class` expression) into an object. This copies all static own properties of the class, but none of the inheritance chain.
- Spreading an iterable (`Map`, `Array`, etc.) into an object. Iterable objects usually do not have meaningful enumerable properties and you probably meant to spread it into an array instead.
```

Reply:
👍 implemented in #10551.